File: app/src/main/java/com/example/pam_app/repository/LanguagesRepository.java
package com.example.pam_app.repository;

import android.content.SharedPreferences;

import java.util.Locale;

public interface LanguagesRepository {

    Locale getCurrentLocale();
    String getKeyPrefLanguage(final String key);
    void setOnSharedPreferencesListener(final SharedPreferences.OnSharedPreferenceChangeListener listener);
    void unregisterOnSharedPreferencesListener();
    void changeLanguage(final String language);
}

File: app/src/main/java/com/example/pam_app/repository/LanguagesRepository.kt
package com.example.pam_app.repository

import android.content.SharedPreferences.OnSharedPreferenceChangeListener
import java.util.*

interface LanguagesRepository {
    var currentLocale: Locale
    fun getKeyPrefLanguage(key: String?): String?
    fun setOnSharedPreferencesListener(listener: OnSharedPreferenceChangeListener?)
    fun unregisterOnSharedPreferencesListener()
    fun changeLanguage(language: String?)
}